src/Hedgehog/Outcome.fs

```diff
@@ -0,0 +1,40 @@
+namespace Hedgehog
+
+type Outcome<'a> =
+    | Failure
+    | Discard
+    | Success of 'a
+
+module Outcome =
```

Comment:
Very optional suggestion.  Instead of matching three times, the duplicate code could be extracted into a function called `cata` function [like Mark Seemann does](https://blog.ploeh.dk/2019/04/29/catamorphisms/).

Reply:
Done